### ISSUE TITLE
Game loads many times while hacking

### ISSUE THREAD

**Issue by lumbar527** [naev/naev#2339]
**Summary**
When doing the Nelly mission to investigate the point of interest, you have to hack into the "Pristine Derelict". While trying to hack into it, Naev pauses every few seconds to load. Clicking is very hard, and it will not always recognize that I am pressing a button. This is not only for clicking on the codes, but for pressing the `Exit to title` button.

**Requested change**
That hacking would run more smoothly? Thanks!

**Versions**
Naev: [0.10.5](https://github.com/naev/naev/releases/tag/v0.10.5)
OS: MacOS 10.15.7 Catalina

**Comment by LJDude** [comment]
What are the specs of your pc, and are you running the game off of an ssd or hdd?

**Comment by lumbar527** [comment]
It's a Mac and I downloaded the game from releases? I'm not entirely sure what you mean.

**Comment by LJDude** [comment]
What is the processor in your mac, and is your storage drive an ssd or an hdd?

**Comment by ProjectSynchro** [comment]
Since we cross-compile our macOS binaries, there might be some loss in performance, but don't quote me on that..
Could you send your save file here? 

I can try and reproduce it on another operating system to verify it's not a lua specific issue.

**Comment by bobbens** [comment]
What version are you on? IIRC we fixed an issue related to this but it hasn't been officially released. Nightly should have the fix though.

**Comment by lumbar527** [comment]
> What version are you on? IIRC we fixed an issue related to this but it hasn't been officially released. Nightly should have the fix though.

[0.10.5](https://github.com/naev/naev/releases/tag/v0.10.5)
So I should download the nightly version.

**Comment by lumbar527** [comment]
> Could you send your save file here?

[James Bubbles.zip](https://github.com/naev/naev/files/11943463/James.Bubbles.zip)
(Just use the autosave.ns)

**Comment by bobbens** [comment]
> > What version are you on? IIRC we fixed an issue related to this but it hasn't been officially released. Nightly should have the fix though.
> 
> [0.10.5](https://github.com/naev/naev/releases/tag/v0.10.5) So I should download the nightly version?

I would back up your save and try the nightly to see if it is fixed. If it is, we should be doing a new release soon which should contain the fix. If not, we'll have to debug more.

**Comment by lumbar527** [comment]
New problem: Nightly will not load my save. I tell it to update it anyway and it starts loading, then I have to force quit.
Screenshots:
<img width="1680" alt="Screen Shot 2023-07-04 at 9 30 40 AM" src="https://github.com/naev/naev/assets/125756761/9bce85e6-0265-4050-9841-3add5c0553e4">
<img width="337" alt="Screen Shot 2023-07-04 at 9 31 24 AM" src="https://github.com/naev/naev/assets/125756761/0789a340-7c62-40ef-8e23-3677881f3636">

**Comment by bobbens** [comment]
OK, I've figured out why your game won't load and solved the issue. It should appear in the next nightly in about 10 hours. You can abort the "Empire Recruitment" mission to get it to work right away though. The issue is that Triton Station has been renamed and isn't found when loading the mission causing it to crash. Thanks for the save!

Still would need to see if nightly solves the POI issue though.

**Comment by lumbar527** [comment]
So if I wait for the next Nightly it should load?

**Comment by bobbens** [comment]
Yes. New nightly should be up now. Remember to back up your save because a lot has changed and you'll get a lot of warning spam due to system renaming / changes when updating.

**Comment by bobbens** [comment]
I just tried it with your save and can't reproduce on nightly... :/
What specs does your computer have? What operating system are you on?

**Comment by lumbar527** [comment]
A 2019 MacBook Pro running Catalina.

**Comment by lumbar527** [comment]
Another thing that might be helpful: when hacking, there's blue spots and lines that move every time it loads. (Sorry, don't have a screenshot yet)

**Comment by bobbens** [comment]
Chances are it's the background shader not playing well with your hardware. Do any of the other minigames work? Or is this the first one you encounter? IIRC the mining minigame does not use the fancy background.

You can also test to see if it gives you issues in the crimson gauntlet system. Try typing `player.teleport("Crimson Gauntlet")` while not landed in the console (open with F2), and it should teleport you there. Does it give issues?

**Comment by lumbar527** [comment]
It might be the first minigame, though I have raced and done the Nelly mining thing. They were both fine. (Give me a second for the teleportation)

**Comment by lumbar527** [comment]
Update: the Crimson Gauntlet is also really slow.

**Comment by bobbens** [comment]
OK, thanks for the confirmation. Make sure your OpenGL drivers are up to date. I have to think of how to deal with this in an elegant way, but for now you can edit `dat/scripts/love_shaders.lua` and find the `love_shaders.circuit` function and edit it to be like below:
```
--[[--
An electronic circuit-board like shader. Meant as/for backgrounds.

@see shaderparams
@tparam @{shaderparams} params Parameter table where "strength" and "speed" fields is used.
--]]
function love_shaders.circuit( params )
   params = params or {}
   local strength = params.strength or 1.0
   local speed = params.speed or 1.0
   local pixelcode = string.format([[
uniform vec3 u_camera = vec3( 0.0, 0.0, 1.0 );
vec4 effect( vec4 color, Image tex, vec2 texture_coords, vec2 screen_coords )
{
   return vec4(0.0) + vec4(u_camera,0.0);
}
]], strength, speed, love_math.random() )

   local shader = graphics.newShader( pixelcode, _vertexcode )
   return shader
end
```
and it should work!

**Comment by lumbar527** [comment]
The plugin works, thanks!

**Comment by bobbens** [comment]
Great! I'll leave this as a plugin then (not in the main game). Hopefully it shouldn't need changes for a while. Thanks for the report! If you find other issues with your hardware, I can try to add fixes either to the main game or the plugin.

**Comment by lumbar527** [comment]
Sorry, related problem. I just tested boarding (had only tested Crimson Gauntlet) and wasn't allowed to hack because I got booted out after the conversation.

**Comment by bobbens** [comment]
Sorry about that, could you try replacing the plugin with the one below? I tried locally and it seems to work now.
[simple-shaders.zip](https://github.com/naev/naev/files/11982754/simple-shaders.zip)

**Comment by lumbar527** [comment]
I didn't finish hacking, but I could easily input letters. Thanks!
